server/rooms.js

```diff
@@ -1693,8 +1693,12 @@ let Rooms = Object.assign(getRoom, {
 				user.inviteOnlyNextBattle = false;
 			}
 		}
-		if (options.tour && !room.tour.modjoin) inviteOnly = [];
-		if (inviteOnly.length) {
+		const prefix = /** @type {RoomBattle} */(room.game).forcedPublic();
+		if (prefix && inviteOnly.length) {
+			room.isPrivate = false;
+			room.modjoin = null;
+			room.add(`|raw|<div class="broadcast-red"><strong>This battle is required to be public due to a player having a name prefixed by '${prefix}'.</div>`);
+		} else if (!options.tour || room.tour.modjoin) {
```

Comment:
This seems to be a faulty if condition? Like, this could run even if `inviteOnly.length` is empty.

Reply:
right, I meant to `if (inviteOnly.length)` the whole thing, thanks.